src/main/java/com/sequenceiq/cloudbreak/service/stack/resource/azure/builders/instance/AzureVirtualMachineResourceBuilder.java

```diff
@@ -199,18 +145,65 @@ public CreateResourceRequest buildCreateRequest(AzureProvisionContextObject prov
         props.put(PORTS, ports);
         props.put(VMTYPE, AzureVmType.valueOf(azureTemplate.getVmType()).vmType().replaceAll(" ", ""));
-        return new AzureVirtualMachineCreateRequest(props, azureStackUtil.createAzureClient(azureCredential), resources, buildResources);
+        return new AzureVirtualMachineCreateRequest(props, azureStackUtil.createAzureClient(azureCredential), resources, buildResources, stack);
+    }
+
+    private String buildimageStoreUri(String commonName, String vmName) {
+        return String.format("http://%s.blob.core.windows.net/vhd-store/%s.vhd", commonName, vmName);
+    }
+
+    @Override
+    public Boolean delete(Resource resource, AzureDeleteContextObject aDCO, String region) throws Exception {
+        Stack stack = stackRepository.findById(aDCO.getStackId());
+        AzureCredential credential = (AzureCredential) stack.getCredential();
+        try {
+            Map<String, String> props = new HashMap<>();
+            props.put(SERVICENAME, resource.getResourceName());
+            props.put(NAME, resource.getResourceName());
+            AzureClient azureClient = azureStackUtil.createAzureClient(credential);
+            HttpResponseDecorator deleteVirtualMachineResult = (HttpResponseDecorator) azureClient.deleteVirtualMachine(props);
+            String requestId = (String) azureClient.getRequestId(deleteVirtualMachineResult);
+            waitUntilComplete(azureClient, requestId);
+        } catch (HttpResponseException ex) {
+            httpResponseExceptionHandler(ex, resource.getResourceName(), stack.getOwner(), stack);
+        } catch (Exception ex) {
+            throw new InternalServerException(ex.getMessage());
+        }
+        return true;
+    }
+
+    @Override
+    public Boolean start(AzureStartStopContextObject aSSCO, Resource resource, String region) {
+        Stack stack = stackRepository.findById(aSSCO.getStack().getId());
+        AzureCredential credential = (AzureCredential) stack.getCredential();
+        boolean started = setStackState(aSSCO.getStack().getId(), resource, azureStackUtil.createAzureClient(credential), false);
+        if (started) {
+            azurePollingService.pollWithTimeout(
+                    new AzureInstanceStatusCheckerTask(),
+                    new AzureInstances(aSSCO.getStack(), azureStackUtil.createAzureClient(credential), Arrays.asList(resource.getResourceName()), "Running"),
+                    POLLING_INTERVAL,
+                    MAX_ATTEMPTS_FOR_AMBARI_OPS);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Boolean stop(AzureStartStopContextObject aSSCO, Resource resource, String region) {
+        Stack stack = stackRepository.findById(aSSCO.getStack().getId());
+        AzureCredential credential = (AzureCredential) stack.getCredential();
+        return setStackState(aSSCO.getStack().getId(), resource, azureStackUtil.createAzureClient(credential), true);
```

Comment:
why don't we wait here until the stop is completed, like in all the other cases?